src/settings/controllers/LazyLoadingController.js

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2018 New Vector
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import SettingsStore from "../SettingsStore";
+import SettingController from "./SettingController";
+import Modal from "../../Modal";
+import sdk from "../../index";
+import MatrixClientPeg  from "../../MatrixClientPeg";
+import dis from "../../dispatcher";
+import { _t } from "../../languageHandler";
+
+export default class LazyLoadingController extends SettingController {
+    onChange(level, roomId, newValue) {
+        dis.dispatch({action: 'flush_storage_reload'});
+    }
+
+    canChangeTo(level, roomId, newValue) {
+        return new Promise((resolve) => this._showReloadDialog(resolve));
```

Comment:
This kinda feels like an abuse of setting controllers: the warning should be hooked up elsewhere in my opinion. The setting controllers are intended to be used for things like desktop notifications where the platform may not support it at all, so the feature should not be enabled.

There's some information about their intent here: https://github.com/matrix-org/matrix-react-sdk/blob/master/docs/settings.md#setting-controllers

Reply:
Ok, I'll do it in the onChange handler for labs checks in UserSettings. As this won't stay a feature for too long hopefully, it's good not to make changes too deep :+1: